### DISCUSSION TITLE
No intellisense in VSCode?

### DISCUSSION THREAD

**Discussion by 0xflux** [microsoft/windows-drivers-rs#176] [General]
**Ignore this, I was using the EWDK. Installing the standard WDK resolved this problem :)**

Hey, I don't know if this is me being stupid, it's my first driver in Rust; but i have no intellisense through VSCode. It may be a configuration problem with VSCode; but given thats my daily driver for Rust I'm not sure it is? I assume the norm is full intellisense? Or is this a known byproduct of the crate?

I've attached a screenshot of my IDE, looking for `DriverUnload()`. `DRIVER_OBJECT` also cant be found by VSCode, either can `PCUNICODE_STRING`.

Many thanks in advance.

![image](https://github.com/microsoft/windows-drivers-rs/assets/49762827/6b0fc96c-70d6-4357-8e01-da7fbd2654b9)

**lib.rs:**

```Rust
#![no_std]

use wdk_sys::ntddk::DbgPrint;

use wdk_sys::{DRIVER_OBJECT, NTSTATUS, PCUNICODE_STRING};

#[cfg(not(test))]
extern crate wdk_panic;

#[cfg(not(test))]
use wdk_alloc::WDKAllocator;

#[cfg(not(test))]
#[global_allocator]
static GLOBAL_ALLOCATOR: WDKAllocator = WDKAllocator;

#[export_name = "DriverEntry"]
pub unsafe extern "system" fn driver_entry(
    driver: &mut DRIVER_OBJECT,
    registry_path: PCUNICODE_STRING,
) -> NTSTATUS {
    DbgPrint("Sanctum driver starting..".as_ptr() as *const i8);

    driver.D

    0
}
```

**Cargo.toml:**

```Rust
[package]
name = "sanctum-driver"
version = "0.0.1"
edition = "2021"
authors = ["Flux Sec <fluxsec@proton.me>"]

[lib]
crate-type = ["cdylib"]
test = false

[package.metadata.wdk]

[profile.dev]
panic = "abort"
lto = true

[profile.release]
panic = "abort"
lto = true

[dependencies]
wdk = { version = "0.2.0", features = [] }
wdk-alloc = "0.2.0"
wdk-panic = "0.2.0"
wdk-sys = "0.2.0"

[build-dependencies]
wdk-build = "0.2.0"

[features]
default = []
nightly = ["wdk/nightly", "wdk-sys/nightly"]
```

**Comment by 0xflux** [comment]
Ignore this, I was using the EWDK. Installing the standard WDK resolved this problem :)

**Reply by wmmc88** [reply]
Intellisense also works for eWDK, if you start vscode from an eWDK prompt

**Reply by 0xflux** [reply]
Awesome! Didn't know that, thanks for clarifying :)
